quipucords/scanner/openshift/api.py

```diff
@@ -138,6 +139,54 @@ def can_connect(self, raise_exception=False, **kwargs):
             return False
         return True
 
+    # Adding support for fetching OCP Prometheus Metrics
+    # Reference: https://access.redhat.com/solutions/3775611
+
+    def metrics_query(self, query):
+        """Execute an OpenShift Prometheus Query."""
+        kube_config = self._configuration
+        if not self._metrics_host:
+            return []
+        try:
+            response = self._api_client.request(
+                "GET",
+                url=f"https://{self._metrics_host}/api/v1/query?query={query}",
+                headers=kube_config.api_key,
+            )
+        except ApiException as err:
+            logger.error(
+                "Failed to execute the Openshift metrics query: '%s' error: '%s'",
+                query,
+                err.reason,
+            )
+            return []
```

Comment:
consider wrapping this function with `catch_k8s_exception` instead as it is our "standard way" to deal with exceptions on ocp API calls.

We could improve how errors are logged there.

Reply:
I was using catch_k8s_exception, but other than the logging, I did not want the error to raise an exception but rather just log the error/query, and return [] to be minimally intrusive.